### ISSUE TITLE
Add 3 octave scales shapes to renderScale 

### ISSUE THREAD

**Issue by Aduffy** [moonwave99/fretboard.js#23]
Compliments on the new renderScale in .2! It is friggin brilliant. I implemented it to replace my code that was using your previous merged boxes examples.. SIGNIFICANTLY less code! 

Do you have plans to add additional box shapes such as 3 octave scales ?

**Comment by moonwave99** [comment]
I think that more extended patterns are:

1. hard to abstract;
2. basically composition of existing ones.

I exclude an automatic generation for the moment.

A decent compromise for the time being is to merge consecutive boxes, filtering out the notes you do not need. For instance for a 3 octaves G major scale, you can glue:

- G major (E shape) - G2 s6:f3 to G3 s4:f5;
- G major (C shape)  - A3 s4:f7 up to D5 s1:f10;
- G major (G shape) - E5 s1:f12 to G5 s1:f15.

It shouldn't be complicated to write an helper that gets the box, filters out the notes lower than the threshold, and concatenates everything ^^

---

I aim at generating small experiments / playgrounds / exercises more than at having an all-inclusive library - mainly because I should be focusing more on playing guitar than implementing features here ^____________^

**Comment by Aduffy** [comment]
Fair enough.. and ditto.. I also should be focusing more on playing and less on coding!! :P

**Comment by Aduffy** [comment]
Are box for 3NPS and CAGED for Melodic Minor and Harmonic Minor scales on your road map at all? 

![image](https://user-images.githubusercontent.com/265857/103471834-fb541b80-4d39-11eb-99e7-9f80c6ee03b8.png)
![image](https://user-images.githubusercontent.com/265857/103471836-19218080-4d3a-11eb-8756-470298aa683b.png)

**Comment by moonwave99** [comment]
Should not be much complicated because they are one position away from the existing TNPS patterns.  
The only problem maybe the mode numbers, since they differ from the major scale and have quite exotic names.

I will check it somewhen in the next days, and if possible will implement it right away ^^
